### ISSUE TITLE
Node cost calculation is arbitrary and obscure

### ISSUE THREAD

**Issue by heijul** [heijul/pdf2gtfs#3]
There are a few issues with how the node costs are calculated.

1. First and foremost, the function currently used, is arbitrary at best.\
(Currently uses `(route_type_score + node_has_this_many_optional_values) ** 2 // 20`)
2. We should also change where the node costs are calculated, instead of doing something like [this](https://github.com/heijul/pdf2gtfs/blob/93cb19541265c5574ff26f3277be9999d5880b0d/src/pdf2gtfs/locate/__init__.py#L195).

In other words, we probably need to 
- [ ] Move everything node cost related to a single place
- [ ] Find a function that has some fixed limit and increases fast for smaller values. (log?)
- [ ] Document how the function works and why it is a good function
